uktrade/data-hub-frontend
# 'Sort by' feature on Contact Activity

## Description of change

We [previously introduced a feature-flag-protected component](https://github.com/uktrade/data-hub-frontend/issues?q=label%3Aaventri+is%3Aclosed) on Contacts that pulls activities from Activity Stream for that contact.

We now want to sort Contact Activities by 'newest' or 'oldest' - based on
when the activity was published. This might not be the final sorting
methods we use, but as we only have limited data on activities for
Aventri this makes sense for now.

notes: 
At some point we will want to do a refactor so that Activity Feed stuff is
not under the 'companies' folder, as it's being used across Contacts and
in the future probably Events, but that could be in a future PR.

## feedback
* naming/ any tidying up we can do
* are we using the routed stuff right? we push the param to history and then the `state2props` checks it and updates state, causing a re-render?


## Test instructions

* ensure you have the user feature flag enabled for your user -> add to your adviser in Django admin `user-contact-activities`
* visit http://localhost:3000/contacts/68be55fc-544d-4d9c-a264-977b917f1fd3/interactions and note that you can use the sort by. 
* it should add params to the url which mean you can copy that link and it determines the page/sortby

## Screenshots
### Before


![Screenshot 2022-05-25 at 12 03 54](https://user-images.githubusercontent.com/16647486/170248409-cda7ef45-b432-4786-9681-a00162d2ee62.png)


### After

![Screenshot 2022-05-25 at 12 04 01](https://user-images.githubusercontent.com/16647486/170248387-d055e155-2e26-41e2-858b-10d0a04982be.png)


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
